extras/__init__.py

```diff
@@ -40,22 +40,27 @@ def try_import(name, alternative=None, error_callback=None):
     """
     module_segments = name.split('.')
     last_error = None
+    remainder = []
+    # module_name will be what imported. We cannot walk from the __import__
+    # result because in import loops (A imports A.B imports C calls
+    # try_import(A.B)) A.B will not yet be set.
```

Comment:
This comment doesn't make sense to me... I think it should be:

"module_name will be what _was_ imported"?

The parenthetical could also be clearer I think:

"(A imports A.B, which imports C, which calls try_import(A.B))"